src/m2n/PointToPointCommunication.cpp

```diff
@@ -496,6 +566,73 @@ void PointToPointCommunication::requestConnection(std::string const &acceptorNam
   _isConnected = true;
 }
 
+void PointToPointCommunication::requestPreConnection(std::string const &acceptorName,
+                                                     std::string const &requesterName)
+{
+  TRACE(acceptorName, requesterName);
+  CHECK(not isConnected(), "Already connected!");
+  CHECK(utils::MasterSlave::isMaster() || utils::MasterSlave::isSlave(),
+        "You can only use a point-to-point communication between two participants which both use a master. "
+        << "Please use distribution-type gather-scatter instead.");
+
+  if (utils::MasterSlave::isMaster()) {
+    // Establish connection between participants' master processes.
+    auto c = _communicationFactory->newCommunication();
```

Comment:
Do we actually need this master-master connection? Nothing is exchanged here, right?

Reply:
removed!